uaclient/entitlements/fips.py

```diff
@@ -153,8 +154,12 @@ def _replace_metapackage_on_cloud_instance(
         if series != "bionic":
             return packages
 
-        cloud_id = get_cloud_type()
-        if not cloud_id or cloud_id not in ("azure", "aws"):
+        cloud_match = re.match(
+            r"^(?P<cloud>(azure|aws)).*", get_cloud_type() or ""
+        )
+        cloud_id = cloud_match.group("cloud") if cloud_match else ""
+
+        if cloud_id not in ("azure", "aws"):
```

Comment:
Since we weren't strongly on the fence one way or another.

Let's stick with the regex because it's a more strict compound check in "one" operation. Though it ultimately takes more lines to express that.